File: practice/convert/example2_book_library/java/book-library/src/main/java/dev/starryeye/book_library/controller/book/LoanBookController.java
package dev.starryeye.book_library.controller.book;

import dev.starryeye.book_library.application.facade.LoanBookFacade;
import dev.starryeye.book_library.controller.book.request.LoanBookRequest;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
public class LoanBookController {

    private final LoanBookFacade facade;

    @PostMapping("/api/v1/books/loan")
    public void loanBook(@Valid @RequestBody LoanBookRequest request) {
        facade.loan(request.toInput());
    }
}

File: practice/convert/example2_book_library/kotlin/book-library/src/main/kotlin/dev/starryeye/book_library/controller/book/LoanBookController.kt
package dev.starryeye.book_library.controller.book

import dev.starryeye.book_library.application.facade.LoanBookFacade
import dev.starryeye.book_library.controller.book.request.LoanBookRequest
import dev.starryeye.book_library.controller.book.request.toInput
import jakarta.validation.Valid
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class LoanBookController(
    private val facade: LoanBookFacade,
) {

    @PostMapping("/api/v1/books/loan")
    fun loanBook(
        @Valid @RequestBody request: LoanBookRequest,
    ) {
        facade.loanBook(request.toInput())
    }
}